src/batArray.mliv

```diff
@@ -533,6 +533,18 @@ val ord : 'a BatOrd.ord -> 'a array BatOrd.ord
     lexicographically for arrays of the same size.  This is a
     different ordering than [compare], but is often faster. *)
 
+val shuffle : ?state:Random.State.t -> 'a array -> 'a array
```

Comment:
Should the shuffle on array really have type `'a array -> 'a array` while it is done in-place (mutate the input and returns it)? I would rather expect `'a array -> unit` to better reflect the behavior. An external opinion would help.